solidity/contracts/bridge/Bridge.sol

```diff
@@ -138,12 +197,69 @@ contract Bridge is Ownable {
     ///         prefix refers to the transaction which created that main UTXO.
+    ///         The txHash is LE bytes32, txOutputIndex an uint32, and
+    ///         txOutputValue an uint64 value.
     mapping(bytes20 => bytes32) public mainUtxos;
 
+    /// @notice Maps the 20-byte wallet public key hash (computed using HASH160
+    ///         opcode) to the total redeemable value of pending redemption
+    ///         requests targeting that wallet. Redeemable value of each request
+    ///         is its requested amount reduced by the treasury fee at the
+    ///         moment of request creation.
+    mapping(bytes20 => uint256) public pendingRedemptionRequestsValue;
+
+    /// @notice Collection of all pending redemption requests indexed by
+    ///         redemption key built as
+    ///         keccak256(walletPubKeyHash | redeemerOutputHash). The
+    ///         walletPubKeyHash is the 20-byte wallet's public key hash
+    ///         (computed using HASH160 opcode) and redeemerOutputHash is the
+    ///         20-byte (P2PKH, P2WPKH or P2SH) or 32-byte (P2WSH) output
+    ///         hash that will be used to lock redeemed BTC as requested by
+    ///         the redeemer. Requests are added to this mapping by the
+    ///         `requestRedemption` method (duplicates not allowed) and are
+    ///         removed by one of the following methods:
+    ///         - `submitRedemptionProof` in case the request was handled
+    ///           successfully
+    ///         - `notifyRedemptionTimeout` in case the request was reported
+    ///           to be timed out
+    ///         - `submitRedemptionFraudProof` in case the request was handled
+    ///           in an incorrect way amount-wise.
+    mapping(uint256 => RedemptionRequest) public pendingRedemptionRequests;
+
+    /// @notice Collection of all redemption faults indexed by redemption key
+    ///         built as keccak256(walletPubKeyHash | redeemerOutputHash). The
+    ///         walletPubKeyHash is the 20-byte wallet's public key hash
+    ///         (computed using HASH160 opcode) and redeemerOutputHash is the
+    ///         20-byte (P2PKH, P2WPKH or P2SH) or 32-byte (P2WSH) output
+    ///         hash that is involved in the fault. Two methods can add
+    ///         to this mapping:
+    ///         - `notifyRedemptionTimeout` which puts the redemption key
+    ///           to this mapping basing on a timed out request stored
+    ///           previously in `pendingRedemptionRequests` mapping.
+    ///         - `submitRedemptionFraudProof` which puts the redemption key
+    ///           to this mapping basing on a misfunded request stored
+    ///           previously in `pendingRedemptionRequests` mapping or
+    ///           basing on a non-requested arbitrary redemption performed
+    ///           by the wallet.
+    // TODO: Remove that Slither disable once this variable is used.
```

Comment:
We could explain in docs why do we store them (to avoid slashing more than one time for the same).

Reply:
See dbb7bf37